src/gluonts/distribution/binned.py

```diff
@@ -157,7 +158,7 @@ def step(p, state):
         a = centers_expanded.pick(idx, axis=-1)
         return a.swapaxes(0, 1)
 
-    def sample(self, num_samples=None):
+    def sample(self, num_samples=None, dtype=np.float32):
```

Comment:
`dtype` isn't used in the function?

Reply:
This distribution does not need the `dtype` because it uses sampling from multinomial distribution which returns integers (`int32`). I added here to be consistent with the signature of `sample` from the abstract class.

We should remove this once [this issue](https://github.com/apache/incubator-mxnet/issues/16407) is fixed on the mxnet side.